src/main/java/com/datadoghq/connect/logs/util/RetryUtil.java

```diff
@@ -0,0 +1,57 @@
+package com.datadoghq.connect.logs.util;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility to compute the retry times for a given attempt, using exponential backoff.
+ * <p>
+ * The purposes of using exponential backoff is to give the Datadog service time to recover when it becomes overwhelmed.
+ * Adding jitter attempts to prevent a thundering herd, where large numbers of requests from many tasks overwhelm the
+ * Datadog service, and without randomization all tasks retry at the same time. Randomization should spread the retries
+ * out and should reduce the overall time required to complete all attempts.
+ */
+public class RetryUtil {
+
+    /**
+     * An arbitrary absolute maximum practical retry time.
+     */
+    public static final long MAX_RETRY_TIME_MS = TimeUnit.HOURS.toMillis(24);
```

Comment:
This value is very big for a max retry time. What do you think about something like 10 minutes?

Reply:
Makes sense to me. I do agree it's long.